aggregator/src/aggregator/aggregation_job_driver.rs

```diff
@@ -236,9 +236,9 @@ impl AggregationJobDriver {
         for report_aggregation in &report_aggregations {
             match report_aggregation.state() {
                 ReportAggregationState::Start => saw_start = true,
-                ReportAggregationState::Waiting(_) => saw_waiting = true,
+                ReportAggregationState::WaitingLeader(_) => saw_waiting = true,
                 ReportAggregationState::Finished => saw_finished = true,
-                ReportAggregationState::Failed(_) => (), // ignore failed aggregations
+                _ => (), // ignore failed aggregations
```

Comment:
Should we return an error (or maybe panic, though I think returning an error is wiser) if we see `WaitingHelper` in Leader code or vice-versa?